implementation/building.tex

```diff
@@ -0,0 +1,165 @@
+\section{Build System}
+\label{sec:build_system}
+
+Building the library and the EFM32 executables turned out to be one of the major parts of the work done in this project.
+This section describes how building the project have changed over time.
+This includes managing dependencies, compiling the RCL and REL and the Silicon Labs EFM32 \texttt{emlib}.
+We have also utilized a continuous integration system that has helped us to keep the project up to date with the nightly builds of Rust and making sure that the builds have been consistent across the systems it has been built on.
+
```

Comment:
Include a birds eye view of the migration process:

`Makefile (rustc+gcc) =>  ... => cargo+linkargs => cargo rustc`